crypto/params.c

```diff
@@ -868,7 +869,7 @@ int OSSL_PARAM_get_uint64(const OSSL_PARAM *p, uint64_t *val)
         return 0;
     }
 
-    if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER) {
+    if (ossl_likely(p->data_type == OSSL_PARAM_UNSIGNED_INTEGER)) {
```

Comment:
There are a lot of similar ones in this file that aren't updated.  I'd suggest staying consistent and doing all or none of these checks.

Reply:
That's right, but the data says different. (Probably will need to updated the description and add more info, but for now tl;dr:) I checked the coveralls report, to check for hot-paths (depending on our test suite) and focused only on the most used lines. If you check [this](https://coveralls.io/builds/74456755/source?filename=crypto%2Fparams.c#L871) file, no other places have that huge usage as this one. I'm focusing only on the most used places where this optimization can really matter.

Reply:
I'm not sure our CI tests are all that representative or normal use.

Reply:
I agree that the test suite is not completely showing true usage, but it could show at least some data. It is very hard to get high amount of real data on this.